specs/Extensions.md

```diff
@@ -0,0 +1,579 @@
+
+### Win32 C++
+```cpp
+static constexpr WCHAR m_defaultExtensionFolderPath = L"//Mydrive";
+
+void AppWindow::InitializeWebView()
+{
+    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+    CHECK_FAILURE(options->put_AreBrowserExtensionsEnabled(TRUE));
+
+    // ... other option properties
+
+    // ... CreateCoreWebView2EnvironmentWithOptions
+
+    InstallDefaultExtensions();
+    
+    // ... Navigate
+}
+
+void ScenarioExtensionsManagement::InstallDefaultExtensions()
+{
+    auto webView2_13 = m_webView.try_query<ICoreWebView2_13>();
+    CHECK_FEATURE_RETURN_EMPTY(webView2_13);
+    wil::com_ptr<ICoreWebView2Profile> webView2Profile;
+    CHECK_FAILURE(webView2_13->get_Profile(&webView2Profile));
+    auto profile6 = webView2Profile.try_query<ICoreWebView2Profile6>();
+    CHECK_FEATURE_RETURN_EMPTY(profile6);
+
+    profile6->GetBrowserExtensions(
+        Callback<ICoreWebView2ProfileGetBrowserExtensionCompletedHandler>(
+            [this, profile6, extension_path](
+                HRESULT error, ICoreWebView2BrowserExtensionList* extensions) -> HRESULT
+            {
+                std::wstring extensionIdString;
+                bool extensionInstalled = false;
+                UINT extensionsCount = 0;
+                extensions->get_Count(&extensionsCount);
+
+                for (UINT index = 0; index < extensionsCount; ++index)
+                {
+                    wil::com_ptr<ICoreWebView2BrowserExtension> extension;
+                    extensions->GetValueAtIndex(index, &extension);
+
+                    wil::unique_cotaskmem_string id;
+                    std::wstring message;
+
+                    extension->get_Id(&id);
+                    extensionIdString = id.get();
+
+                    if (extensionIdString.compare(m_extensionId) == 0)
+                    {
+                        extensionInstalled = true;
+                        break;
+                    }
+                }
+
+                if (!extensionInstalled)
+                {
+                    CHECK_FAILURE(profile6->AddBrowserExtension(
+                        m_defaultExtensionFolderPath.c_str(),
+                        Callback<ICoreWebView2ProfileAddBrowserExtensionCompletedHandler>(
+                            [](HRESULT error,
+                               ICoreWebView2BrowserExtension* extension) -> HRESULT
+                            {
+                                if (error != S_OK)
+                                {
+                                    ShowFailure(error, L"AddExtension failed");
+                                    return S_OK;
+                                }
+                                return S_OK;
+                            })
+                            .Get()));
+                }
+                return S_OK;
+            })
+            .Get());
+}
+```
+
+### .NET and WinRT
+```c#
+string m_defaultExtensionId = "123";
```

Comment:
Please make these two more realistic values and add comments saying you need to change them to match the package ID and path of the package you want to install.